spec/lib/jobs/dell-wsman-simple-update-firmware-spec.js

```diff
@@ -0,0 +1,278 @@
+
+'use strict';
+
+var uuid = require('node-uuid');
+
+describe(require('path').basename(__filename), function() {
+    var SimpleUpdateFirmwareJob;
+    var simpleUpdateFirmwareJob;
+    var wsmanBaseJob;
+    var wsmanTool;
+    var configuration;
+    var logger;
+    var errorLoggerSpy;
+    var fs;
+    var ChildProcess;
+
+    var obm = {
+        id: '59c4797e70bf1d7c172930dc',
+        node: '/api/2.0/nodes/59c4720870bf1d7c172930db',
+        service: 'dell-wsman-obm-service',
+        config: {
+            user: 'admin',
+            password: 'admin',
+            host: '172.00.00.00'
+        }
+    };
+
+    var dellConfigs = {
+        gateway: 'http://localhost:46010',
+        services: {
+            firmware: {
+                dupUpdater: 'http://localhost:46010/api/1.0/server/firmware/updater/dup'
+            }
+        }
+    };
+
+    before(function() {
+        helper.setupInjector([
+            helper.require('/lib/utils/job-utils/wsman-tool'),
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/jobs/dell-wsman-base-job.js'),
+            helper.require('/lib/jobs/dell-wsman-simple-update-firmware.js')
+        ]);
+        SimpleUpdateFirmwareJob = helper.injector.get('Job.Dell.Wsman.Simple.Update.Firmware');
+        var options = {
+                imageURI: 'http://localhost:8080/common/iDRAC-with-Lifecycle-Controller_Firmware_278FC_WN32_2.50.50.50_A00.EXE'
+            },
+            context = {
+                graphId: 'fdc41b55-d61d-45c1-9b67-651922e94daa'
+            };
+        simpleUpdateFirmwareJob = new SimpleUpdateFirmwareJob(options, context, uuid.v4());
+        wsmanBaseJob = helper.injector.get('Job.Dell.Wsman.Base');
+        wsmanTool = helper.injector.get('JobUtils.WsmanTool');
+        configuration = helper.injector.get('Services.Configuration');
+        fs = helper.injector.get('fs');
+        ChildProcess = helper.injector.get('ChildProcess');
+        logger = helper.injector.get('Logger');
+        errorLoggerSpy = sinon.spy(logger.prototype, 'error');
+        this.sandbox = sinon.sandbox.create();
+    });
+
+    beforeEach(function() {
+        this.sandbox.stub(wsmanBaseJob.prototype, 'checkOBM');
+        this.sandbox.stub(wsmanTool.prototype, 'clientRequest');
+        this.sandbox.stub(fs, 'readFile');
+        this.sandbox.stub(fs, 'existsSync');
+        this.sandbox.stub(ChildProcess.prototype, 'run');
+    });
+
+    afterEach(function() {
+        this.sandbox.restore();
+    });
+
+    describe('_initJob', function() {
+        beforeEach(function() {
+            this.sandbox.stub(configuration, 'get');
+            this.sandbox.stub(SimpleUpdateFirmwareJob.prototype, 'getComponentType').resolves();
+        });
+
+        it('should handle the case of lacking dell configuration', function() {
+            configuration.get.withArgs('dell').returns(undefined);
+            expect(simpleUpdateFirmwareJob._initJob()).to
```

Comment:
The promise need be returned, otherwise this test case will always succeed no matter the promise is rejected or fulfilled.